### PR TITLE
External Kubernetes secrets storage - best practices

### PR DESCRIPTION
**Pull request by katarzyna-koltun-mx**

### CODE REVIEW

**Comment by katarzyna-koltun-mx** [comment]
@Nidhi251289 @zlogic Here's the first PR (new document). Let me know any comments or corrections.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-secrets.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+## 3 Sample Implementations
+
+The following sections outline the process of implementing an external secret store with Vault and with AWS. You can refer to them as an example, and to help you troubleshoot your own implementation.
+
+### 3.1 Configuring a Secret Store with Vault
+
+To enable your environment to use Vault as external secret storage, follow these steps:
+1. Install [Vault](https://developer.hashicorp.com/vault/docs/platform/k8s/helm) and its [CSI Secrets Driver](https://github.com/hashicorp/vault-csi-provider), if it is not already installed in the cluster.
+2. Create a database secret in Vault, as shown in the following example. Replace `<{env-db-secret}>` with a unique name, and update any values to match your database configuration:
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-secrets.md]: from document https://mendix.atlassian.net/wiki/spaces/PCDocs/pages/2150170783/External+Secret+Store-+Vault+Support, can you also add point 2 and 3 from section Vault configuration

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-secrets.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        ]
+    }
+    ```
+9. Create a Kubernetes `ServiceAccount` for your environment:
+    ```shell
+    kubectl -n <{k8s namespace}> create serviceaccount <{environment name}>
+    kubectl -n <{k8s namespace}> annotate serviceaccount <{environment name}> privatecloud.mendix.com/environment-account=true
+    kubectl -n <{k8s namespace}> annotate serviceaccount <{environment name}> eks.amazonaws.com/role-arn=<{aws role ARN}>
+    ```
+10. In the Mendix app, set `allowOverrideSecretsWithSecretStoreCSIDriver` to *true*, and make sure that the app does not use storage plans. For more information, see [Additional considerations](#additional-considerations).
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-secrets.md]: can you give an example of mendix app crd, as shown below:

`cat > mendixApp.yaml <<EOF
apiVersion: privatecloud.mendix.com/v1alpha1
kind: MendixApp
metadata:
  name: <internal Name>
spec:
  mendixRuntimeVersion: <Runtime version>
  allowOverrideSecretsWithSecretStoreCSIDriver: true
  replicas: 1
  resources:
    limits:
      cpu: "1"
      memory: 512Mi
    requests:
      cpu: 100m
      memory: 512Mi
  runtime:
    customConfiguration: '{"ScheduledEventExecution":"NONE","MicroflowConstants":"{\"MyFirstModule.MyConstant\":\"Awesome\",\"RestClient.RestServiceUrl\":\"https://go-dummy-app.privatecloud-storage-tls.svc.cluster.local\",\"Atlas_Core.Atlas_Core_Version\":\"3.0.5\"}"}'
    dtapMode: P
    logAutosubscribeLevel: INFO
    runtimeLicense: {}
  runtimeMetricsConfiguration: {}
  sourceURL: oci image url
  sourceVersion: <version of app>
EOF`

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-secrets.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+          - secretKey: "storage-perform-delete"
+            objectName: "storage-perform-delete"
+            secretPath: "secret/<{env-file-secret}>"
+          #- secretKey: "storage-use-ca-certificates"
+          #  objectAlias: "storage-use-ca-certificates"
+          #  secretPath: "secret/<{env-file-secret}>
+          - secretKey: "mx-admin-password"
+            objectName: "mx-admin-password"
+            secretPath: "secret/<{env-configuration-secret}>"
+    ```
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-secrets.md]: Just like point 10 of AWS secret manager, can you also mention that customer needs to create a mendix app cr and add the attribute as mentioned in point 10.

**Pull request merged**